awakened-poe-trade: init at 3.25.102

nezia1 opened the pull request:
## Description of changes

I've added [Awakened POE](https://github.com/SnosMe/awakened-poe-trade) (a Path of Exile trading app for price checking).

I ended up packaging it for Nix using the AppImage, as the build process for the app involves using a build script from upstream that uses `import electron`, which would use the Node.JS electron version and not take advantage of NixOS' dependency system. Upstream changes would likely need to be made to be packaged from source, or there is probably something I don't know (this is one of my first PRs to nixpkgs).

Closes #302672.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nezia1 committed:
maintainers: add nezia

nezia1 committed:
awakened-poe-trade: init at 3.25.102

dotlambda commented on pkgs/by-name/aw/awakened-poe/package.nix:
```diff
@@ -0,0 +1,64 @@
+  src = fetchurl {
+    url = "https://github.com/SnosMe/awakened-poe-trade/releases/download/v${version}/Awakened-PoE-Trade-${version}.AppImage";
+    hash = "sha256-lcdKJ+B8NQmyMsv+76+eeESSrfR/7Mq6svO5VKaoNUY=";
+    name = "${pname}-${version}";
```
```suggestion
```

dotlambda commented on pkgs/by-name/aw/awakened-poe/package.nix:
```diff
@@ -0,0 +1,64 @@
+
+  appimageContents = appimageTools.extractType2 {
+    name = "${pname}-${version}";
+    src = src;
```
```suggestion
    inherit src;
```

dotlambda commented on pkgs/by-name/aw/awakened-poe/package.nix:
```diff
@@ -0,0 +1,64 @@
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin $out/share/${pname} $out/share/applications
```
```suggestion
    mkdir -p $out/bin $out/share/awakened-poe $out/share/applications
```
#277994

dotlambda commented on pkgs/by-name/aw/awakened-poe/package.nix:
```diff
@@ -0,0 +1,64 @@
+    cp -a ${appimageContents}/awakened-poe-trade.desktop $out/share/applications/${pname}.desktop
+    cp -a ${appimageContents}/usr/share/icons $out/share
+
+    substituteInPlace $out/share/applications/${pname}.desktop \
+    --replace 'Exec=AppRun' 'Exec=${pname}'
```
```suggestion
    substituteInPlace $out/share/applications/awakened-poe.desktop \
    --replace-fail 'Exec=AppRun' 'Exec=awakened-poe'
```

dotlambda commented on pkgs/by-name/aw/awakened-poe/package.nix:
```diff
@@ -0,0 +1,64 @@
+    runHook preInstall
+    mkdir -p $out/bin $out/share/${pname} $out/share/applications
+
+    cp -a ${appimageContents}/{locales,resources} $out/share/${pname}
+    cp -a ${appimageContents}/awakened-poe-trade.desktop $out/share/applications/${pname}.desktop
```
Why don't we use `awakened-poe-trade`? That's also what it's called in the AUR: https://repology.org/project/awakened-poe-trade/versions
```suggestion
    cp -a ${appimageContents}/{locales,resources} $out/share/awakened-poe
    cp -a ${appimageContents}/awakened-poe-trade.desktop $out/share/applications/awakened-poe.desktop
```

dotlambda commented on pkgs/by-name/aw/awakened-poe/package.nix:
```diff
@@ -0,0 +1,64 @@
+  '';
+
+  postFixup = ''
+        makeWrapper ${electron}/bin/electron $out/bin/${pname} \
```
```suggestion
        makeWrapper ${lib.getExe electron} $out/bin/awakened-poe \
```

dotlambda commented on pkgs/by-name/aw/awakened-poe/package.nix:
```diff
@@ -0,0 +1,64 @@
+
+  postFixup = ''
+        makeWrapper ${electron}/bin/electron $out/bin/${pname} \
+    --add-flags $out/share/${pname}/resources/app.asar \
```
```suggestion
    --add-flags $out/share/awakened-poe/resources/app.asar \
```

dotlambda commented on pkgs/by-name/aw/awakened-poe/package.nix:
```diff
@@ -0,0 +1,64 @@
+    license = lib.licenses.mit;
+    maintainers = lib.maintainers.nezia;
+    platforms = lib.platforms.linux;
+    mainProgram = "${pname}";
```
```suggestion
    mainProgram = "awakened-poe";
```

dotlambda commented on pkgs/by-name/aw/awakened-poe/package.nix:
```diff
@@ -0,0 +1,64 @@
+    description = "Path of Exile trading app for price checking";
+    homepage = "https://github.com/SnosMe/awakened-poe-trade";
+    license = lib.licenses.mit;
+    maintainers = lib.maintainers.nezia;
```
```suggestion
    maintainers = [ lib.maintainers.nezia ];
```

dotlambda reviewed (commented):
`meta.changelog` and `meta.sourceProvenance` missing

nezia1 commented:
Renamed the package to go under `pkgs/by-name/aw/awakened-poe-trade`, and renamed all executables and desktop files accordingly. I think you're right and we should keep it consistent with other maintainers. Also added the meta you told me to add (force pushed and squished commits).

dotlambda commented:
Commit message and PR title are wrong.

dotlambda commented on pkgs/by-name/aw/awakened-poe-trade/package.nix:
```diff
@@ -0,0 +1,66 @@
+    description = "Path of Exile trading app for price checking";
+    homepage = "https://github.com/SnosMe/awakened-poe-trade";
+    changelog = "https://github.com/SnosMe/awakened-poe-trade/releases/tag/v${version}";
+    sourceProvenance = [ lib.sourceTypes.binaryBytecode ];
```
What about it is bytecode?

nezia1 replied:
Side-note: should I ignore this warning?
```
❯ ./result/bin/awakened-poe-trade
hook_thread_proc [101]: Could not set thread priority 49 for thread 0x7F0BBEC7F6C0!
load_input_helper [1827]: XkbGetKeyboard failed to locate a valid keyboard!
APPIMAGE env is not defined, current application is not an AppImage
```

dotlambda reviewed (commented):
#277994
